### ISSUE TITLE
JSS Version change in 9.99.0-t1494340586 breaks ruby-jss gem

### ISSUE THREAD

**Issue by csfjeff** [PixarAnimationStudios/ruby-jss#17]
I just updated to the newest version of JSS, 9.99.0-t1494340586, and it looks like it breaks this gem :(.  When I try to create a new JSS::API.connect I get the following error:

/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0.rubygems/version.rb:191:in 'initialize': Malformed version number string 9.9.9.0-t1494340586 (ArgumentError)

**Comment by glenfarclas17** [comment]
Thanks for reporting this!  We've not gone to 9.99 yet, but I've got a fix in the works here. I'll get it pushed to github within a few days!

**Comment by csfjeff** [comment]
Awesome! Thank you!

**Comment by glenfarclas17** [comment]
Just pushed v0.8.1 which should address the new versioning.  Could you give it a try and let me know what you see?  Cheers!

**Comment by csfjeff** [comment]
Just pulled the update and tested on 2 different machines, unfortunately I still seem to get the same error message.  Will paste in the full error just in case that helps:

```
JSS::API.connect(:user=>"username", :pw=>"password", :server=>"serverdnsname")
ArgumentError: Malformed version number string 9.99.0-t1494340586
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/version.rb:191:in `initialize'
	from /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/utility.rb:437:in `new'
	from /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/utility.rb:437:in `parse_jss_version_newstyle'
	from /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/utility.rb:422:in `parse_jss_version'
	from /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/server.rb:90:in `initialize'
	from /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/api_connection.rb:413:in `new'
	from /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/api_connection.rb:413:in `verify_server_version'
	from /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/api_connection.rb:176:in `connect'
	from (irb):2
	from /usr/bin/irb:12:in `<main>'
```

**Comment by glenfarclas17** [comment]
Grr - I tested the version parsing with that specific version  but clearly not enough or in enough context. 

What version of rubygems are you using?  I wonder if the Gem::Version class is behaving differently

**Comment by glenfarclas17** [comment]
@csfjeff  could you try this to see if it fixes in your environment?:

in file /Library/Ruby/Gems/2.0.0/gems/ruby-jss-0.8.1/lib/jss/utility.rb   change line 437
from this:
   version: Gem::Version.new(version)
to this:
   version: Gem::Version.new("#{major}.#{minor}.#{revision}#{build}")

and see if it parses correctly? 

I'm suspecting your version of Gem::Version doesn't know how to handle the '-' in the version number

If so, I'll push a fix asap.

**Comment by csfjeff** [comment]
Will try that first thing in the morning - I broke ruby on my personal laptop with some other dev related thing and haven't had a chance to fix it yet.  Thanks for your efforts btw, much appreciated!

**Comment by csfjeff** [comment]
I'm running my testing on 10.12.5, looks like rubygems is version 2.0.14.1, which I think is the regular version that comes with Sierra.  And good news, I just tested with making the change you requested and that seems to do the trick :).

**Comment by glenfarclas17** [comment]
Thanks so much! 

fix has been pushed as v0.8.2

Let me know if you find other problems!
Cheers,
-Chris

**Comment by csfjeff** [comment]
Awesome! Just tested again with the 0.8.2 release and looks good.  Thanks again!!
